dash-renderer/src/APIController.react.js

```diff
@@ -65,11 +82,24 @@ const UnconnectedContainer = props => {
         content = <div className="_dash-error">Error loading dependencies</div>;
     } else if (appLifecycle === getAppState('HYDRATED')) {
         renderedTree.current = true;
+
         content = (
-            <TreeContainer
-                _dashprivate_layout={layout}
-                _dashprivate_path={[]}
-            />
+            <DashContext.Provider value={provider.current}>
+                <TreeContainer
+                    _dashprivate_error={error}
+                    _dashprivate_layout={layout}
+                    _dashprivate_loadingState={getLoadingState(
+                        layout,
+                        [],
+                        loadingMap
+                    )}
+                    _dashprivate_loadingStateHash={getLoadingHash(
+                        [],
+                        loadingMap
+                    )}
+                    _dashprivate_path={JSON.stringify([])}
```

Comment:
The path of this component inside the layout. As this is built on each render, it will never be the same as for the previous render. Stringify it. It will be handled in the TreeContainer. This with the above modifications allows the TreeContainer to be a PureComponent.

Reply:
I suppose your intent was to be explicit and consistent about always stringifying / parsing when you set / read this value... looks a little weird to me to vs just `_dashprivate_path="[]"` but OK.